src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -0,0 +1,716 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+#if UNIFIED
+using CoreFoundation;
+using Foundation;
+using Security;
+#else
+using MonoTouch.CoreFoundation;
+using MonoTouch.Foundation;
+using MonoTouch.Security;
+using System.Globalization;
+#endif
+
+#if SYSTEM_NET_HTTP
+namespace System.Net.Http 
+#else
+namespace Foundation
+#endif
+{
+	public class NSUrlSessionHandler : HttpClientHandler

```

Comment:
That was all because of `DidReceiveChallenge` and `WillPerformHttpRedirection`.  If you don't think either of those are needed, then I can drop to `HttpMessageHandler` without any objections.